src/producer.rs

```diff
@@ -260,6 +284,20 @@ impl Archive {
 
                 true
             }
+            ArchiveType::Plain(_) => {
+                // don't use a hard link here because it can fail when src and dst are not on the same device
```

Comment:
We should never be in this branch, given that Plain is only for info files, right? If so, just remove all this code and add an assertion failing here